app/models/creditsandrefunds/CreditAndRefundViewModel.scala

```diff
@@ -16,18 +16,35 @@
 
 package models.creditsandrefunds
 
-import models.creditDetailModel.{BalancingChargeCreditType, CreditType, CutOverCreditType, MfaCreditType}
-import models.financialDetails.{DocumentDetailWithDueDate, FinancialDetail}
+import models.financialDetails._
+import play.api.i18n.Messages
 
+import java.time.LocalDate
 
-case class CreditAndRefundViewModel(creditCharges: List[(DocumentDetailWithDueDate, FinancialDetail)]) {
-
+sealed trait CreditRow {
+  val amount: BigDecimal
+  val creditType: CreditType
+}
+case class CreditViewRow(amount: BigDecimal, creditType: CreditType, taxYear: String) extends CreditRow
+case class PaymentCreditRow(amount: BigDecimal, creditType: CreditType, date: LocalDate) extends CreditRow
 
-  private val balancingChargeCredit = "BCC"
-  private val mfaCredit = "MFA"
-  private val cutOverCredit = "CutOver"
-  private val payment = "Payment"
+case class CreditAndRefundViewModel(creditCharges: List[(DocumentDetailWithDueDate, FinancialDetail)]) {
 
+  def creditViewRows()(implicit messages: Messages): List[CreditRow] = {
```

Comment:
generally we avoiding usage of tuple directly as its hard to read this code:
Something like this would improve code readability:

```
val sortedCreditRows: Seq[Option[CreditRow]] = sortCreditsByYear.map(cc =>
    {
      val (documentDetails, financialDetail) = cc
      val maybeCreditRow = for {
        creditType <- financialDetail.getCreditType
        amount <- documentDetails.documentDetail.paymentOrChargeCredit
      } yield {
        creditType match {
          case PaymentType => PaymentCreditRow(amount = amount, creditType = creditType, date =documentDetails.dueDate.get)
          case _ =>  CreditViewRow(amount = amount, creditType = creditType, taxYear = financialDetail.taxYear)
        }
      }
```